[Feature] Support pinot date time function transform

maggie-zhu opened the pull request:
## Why I'm doing:
We have many pinot sqls in the prod environment when we try to migrate the pinot to starrocks. We hope the starrocks can support the sql transformation automatically.
## What I'm doing:

To support the sql transformation from pinot sql to starrocks sql when use some date functions.
## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.4
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

maggie-zhu committed:
[Enhancement] Load and Export add warehouse parameter (#54190)

Signed-off-by: magzhu <magzhu@cisco.com>

maggie-zhu committed:
support pinot dialect

Signed-off-by: magzhu <magzhu@cisco.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/StarRocks/starrocks?pullRequest=54227) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks?pullRequest=54227) before we can accept your contribution.<br/><hr/>**magzhu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/StarRocks/starrocks?pullRequest=54227) it.</sub>

stephen-shelby committed:
[Enhancement] Eliminate sort column when all predicates of the column are equivalent predicates (#54177)

Signed-off-by: stephen <stephen5217@163.com>

Youngwb committed:
[BugFix] Fix query error for delta lake partitioned table with column mapping (#54204)

wyb committed:
[Enhancement] Change insert match by name syntax (#54120)

Signed-off-by: wyb <wybb86@gmail.com>

stephen-shelby committed:
[Enhancement] support collecting array/map type column statistics (#53862)

Signed-off-by: stephen <stephen5217@163.com>

HangyuanLiu committed:
[BugFix] Temporarily compatible with Commit/Rollback Statement during the development phase (#54229)

mxdzs0612 committed:
[Enhancement] Support analyze paimon table and optimize query execution (#52858)

Signed-off-by: Jiao Mingye <mxdzs0612@gmail.com>

luohaha committed:
[Feature] bulk load optimization (Part 2 - SpillMemTableSink) (#53973)

Signed-off-by: luohaha <18810541851@163.com>

luohaha committed:
[BugFix] fix char padding missing after partial update (#54182)

Signed-off-by: luohaha <18810541851@163.com>

starrocks-xupeng committed:
[Enhancement] support lake_clear_corrupted_cache option for BE (#54216)

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>

wyb committed:
[Enhancement] Improve file sample in files() schema detection (#54201)

Signed-off-by: wyb <wybb86@gmail.com>

trueeyu committed:
[Enhancement] RuntimeColumnPredicateBuilder support processing rf with null (#54222)

Signed-off-by: trueeyu <lxhhust350@qq.com>

kalyanmysore committed:
[Doc] Added missing _ and clarifications (#54241)

Signed-off-by: kalyanmysore <kalyanmysore@yahoo.com>

decster committed:
[BugFix] Add log when txn publish start for debugging batch publish (#54255)

Signed-off-by: Binglin Chang <decstery@gmail.com>

sevev committed:
[Enhancement] Support drop cloud native index for primary key table in Shared-data mode (#53971)

Signed-off-by: sevev <qiangzh95@gmail.com>

Youngwb committed:
[Feature] Support translate Trino query to StarRocks Query (#54185)

murphyatwork committed:
[Enhancement] improve table re-create logic of TableKeeper (#54036)

Signed-off-by: Murphy <mofei@starrocks.com>

LiShuMing committed:
[BugFix] Fix possble task run state incosistent in leader/follwer FE (#54225)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

stephen-shelby committed:
[Enhancement] support calcuate unnest table function row nums (#54252)

Signed-off-by: stephen <stephen5217@163.com>

murphyatwork committed:
[Enhancement] optimize buffer strategy of merge-sort (#54183)

Signed-off-by: Murphy <mofei@starrocks.com>

kevincai committed:
[Enhancement] support config s3client request timeout for shared-data fslib (#54211)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

meegoo committed:
[BugFix] Fix load query_mem_tracker leak (#54242)

Signed-off-by: meegoo <meegoo.sr@gmail.com>

gengjun-git committed:
[Enhancement] Change FE compile source/target to 17 (#53617)

1. Copy UnmodifiableCollectionsSerializer.java to our repository to fix compilable issue:
`Unable to make field final java.util.Collection java.util.Collections$UnmodifiableCollection.c accessible: module java.base does not "opens java.util" to unnamed module @61322f9d `
  Refer to https://github.com/magro/kryo-serializers/issues/131
2. Upgrade jacoco to 0.8.8
3. For the arrow flight server, users should add `--add-opens=java.base/java.nio=org.apache.arrow.memory.core,ALL-UNNAMED` to JAVA_OPTS 
  Refer to https://arrow.apache.org/docs/java/install.html
4. Fix some UT

Signed-off-by: gengjun-git <gengjun@starrocks.com>

wyb committed:
[Enhancement] Support list files() recursively (#54256)

Signed-off-by: wyb <wybb86@gmail.com>

starrocks-xupeng committed:
[Doc] adjust data cache doc for 3.4 (#54267)

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

ZiheLiu committed:
[BugFix] Fix arrow flight sql start blcoked (#54295)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

wyb committed:
[Enhancement] Remove csv scanner error report when use flexible column mapping (#54283)

Signed-off-by: wyb <wybb86@gmail.com>

srlch committed:
[UT] remove unstable restore job UT (#54311)

Signed-off-by: srlch <linzichao@starrocks.com>

stdpain committed:
[BugFix] Fix runtime filter crash with aggregate query cache (#54305)

Signed-off-by: stdpain <drfeng08@gmail.com>

stdpain committed:
[Feature] support event scheduler (part 1) (#54260)

Signed-off-by: stdpain <drfeng08@gmail.com>

wxl24life committed:
[Enhancement] Support only deleting shard meta from fe for shared-data cluster (#53141)

Signed-off-by: drake_wang <wxl250059@alibaba-inc.com>

trueeyu committed:
[Enhancement] relax the restriction of  topn runtime filter (#54286)

Signed-off-by: trueeyu <lxhhust350@qq.com>

srlch committed:
[BugFix] Fix restore fail for table with multi level partition (#54314)

Signed-off-by: srlch <linzichao@starrocks.com>

LiShuMing committed:
[BugFix] Fix mv union rewrite and partition prune bug (#54293)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

gengjun-git committed:
[Enhancement] Add some comments to pom.xml (#54331)

1. Add some comment to pom.xml
2. Remove the version configuration from root pom.xml and set it in sub module's pom.xml

Signed-off-by: gengjun-git <gengjun@starrocks.com>

trueeyu committed:
[Enhancement] Relax the restrict of topn runtime filter on iceberg v2 (#54330)

Signed-off-by: trueeyu <lxhhust350@qq.com>

HangyuanLiu committed:
[Enhancement] Fix optimize job fail when use multi warehouse (#54325)

MatthewH00 committed:
[Enhancement] add partition scan num limit when query internal olap table (#53747)

Why I'm doing:
when query big size internal olap table with full table scan or scan too many partitions, would cause BE/CN node high load, lead to cluster instability.

What I'm doing:
add a new FE session variable scan_olap_partition_num_limit to limit partition scan num when query internal olap table.
(default value is 0, means no limitation)

Signed-off-by: MatthewH00 <1639097204@qq.com>
Signed-off-by: hmx <1639097204@qq.com>

LiShuMing committed:
[UT] [Refactor] Fix unstable mv test cases (#54320)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

Youngwb committed:
[Enhancement] add skip rows into split context to avoid re-build deletion vector (#54302)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>

kevincai committed:
[Enhancement] add penalty time when node is added in blocklist (#54203)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

stdpain committed:
[BugFix] Fix query cache crash when enable group execution (#54363)

Signed-off-by: stdpain <drfeng08@gmail.com>

luohaha committed:
[Enhancement] improve cloud native index memtable memory usage and statistic (#54358)

Signed-off-by: luohaha <18810541851@163.com>

LiShuMing committed:
[UT] Fix AggStateCombinatorTest test performance (#54375)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

xiangguangyxg committed:
[Enhancement] Trigger new starmgr image in alter system create image for shared-data mode (#54370)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

hongkunxu committed:
[Doc] Update dynamic partition docs (#54366)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

sevev committed:
[Enhancement] Support remove immutable memtable for cloud native pindex (#54178)

Signed-off-by: sevev <qiangzh95@gmail.com>

murphyatwork committed:
[BugFix] allow partition ttl for statistics database (#54398)

Signed-off-by: Murphy <mofei@starrocks.com>

murphyatwork committed:
[BugFix] check if the table ready to avoid annoying exception (#54365)

Signed-off-by: Murphy <mofei@starrocks.com>

satanson committed:
[UT] Make query cache UT stable (#54410)

Signed-off-by: satanson <ranpanf@gmail.com>

LiShuMing committed:
[BugFix] Fix HistogramMetric output in json format (#54361)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

before-Sunrise committed:
[Enhancement]Support select_if in arm (#53093)

Signed-off-by: before-Sunrise <unclejyj@gmail.com>

qingzhongli committed:
[BugFix] Fix BFD initialization error on aarch64 (#54372)

## Why I'm doing:
BFD initialization error on aarch64:
```
[root@starrocks be]# uname -a
Linux starrocks 4.19.90-2107.6.0.0192.8.oe1.bclinux.aarch64 #1 SMP Tue Mar 21 09:23:05 CST 2023 aarch64 aarch64 aarch64 GNU/Linux
[root@starrocks be]# ./bin/start_be.sh --daemon
[root@starrocks be]# tail log/be.WARNING
E20241217 09:49:19.948438 70369568784448 bfd_parser.cpp:107] set default target to elf64-x86-64 failed.
[root@starrocks be]#
```
## What I'm doing:

Fix BFD initialization error on aarch64.

Signed-off-by: qingzhongli <qingzhongli2018@gmail.com>

xiangguangyxg committed:
[Enhancement] Correct fe node name if needed when fe start with bdbje_reset_election_group (#54399)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

kevincai committed:
[Enhancement] FE interface to allow querying partition meta from backend (#54383)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

srlch committed:
[BugFix] Fix retry upload failure when backup files using HDFS filesystem interface (#53679)

Signed-off-by: srlch <linzichao@starrocks.com>

banmoy committed:
[BugFix] Optimize merge commit performance (#54251)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

xiangguangyxg committed:
[Enhancement] Support empty cluster_snapshot.yaml config file (#54416)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

dirtysalt committed:
[Enhancement] support pk/fk constraint in iceberg table (#54364)

Signed-off-by: yanz <dirtysalt1987@gmail.com>

stdpain committed:
[Enhancement] support exec debug options (#54446)

## Why I'm doing:

Currently it is very difficult to add cases that override the pipeline state with OUTPUT_FULL, this patch implements a debug option to help us add OUTPUT FULL type test cases.

usage:
```
mysql> set query_debug_options = '{"execDebugOptions":[{"plan_node_id":3, "debug_action":"WAIT", "value":500}]}';
Query OK, 0 rows affected (0.00 sec)

mysql> select count(*) from t0;
+----------+
| count(*) |
+----------+
|    40960 |
+----------+
1 row in set (0.51 sec)
```

Signed-off-by: stdpain <drfeng08@gmail.com>

xiangguangyxg committed:
[Feature] Support restoring from a cluster snapshot for shared-data mode (part 3, introduce gtid for tablet metadata) (#54326)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

kevincai committed:
[UT] explicitly ignore surefire maven tests failures in fe-common and hive-udf (#54442)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

kevincai committed:
[Enhancement] validate loop_count_wait_fragments_finish config value (#54395)

luohaha committed:
[BugFix] fix cloud native index memtable memory calculation (#54464)

Signed-off-by: luohaha <18810541851@163.com>

sevev committed:
[UT] Fix unstable ut (#54463)

Signed-off-by: sevev <qiangzh95@gmail.com>

banmoy committed:
[Enhancement] Add storage-related profile for cloud-native (#54007)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

murphyatwork committed:
[Enhancement] improve performance of show materialized views statement (#54374)

Signed-off-by: Murphy <mofei@starrocks.com>

banmoy committed:
[Tool] Set CODEOWNERS of merge commit to load-unload-maintainer (#54490)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

andyziye committed:
[Tool] Change the mode of image update to serial (#54494)

wyb committed:
[Enhancement] Support column mismatch fill property in files() (#54362)

Signed-off-by: wyb <wybb86@gmail.com>

andyziye committed:
[Tool] Fix CI merge pipeline (#54497)

LiShuMing committed:
[Refactor] Refactor mv partition compensate (#54387)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

stdpain committed:
[BugFix] Fix NPE workgroup executor (#54492)

Signed-off-by: stdpain <drfeng08@gmail.com>

srlch committed:
[Feature] Support Cluster Snapshot Backup: SQL Interface and meta data (part 1) (#54447)

Signed-off-by: srlch <linzichao@starrocks.com>

banmoy committed:
[Enhancement] Optimize exception handling for merge commit (#54471)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

stdpain committed:
[BugFix] Use a separate thread pool for dictionary collection (#54454)

Signed-off-by: stdpain <drfeng08@gmail.com>

starrocks-xupeng committed:
[Enhancement] support dummy replication type for StarOSAgent (#54284)

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>

andyziye committed:
[Tool] Fix: Update image even if the previous task failed (#54500)

Signed-off-by: andyziye <108652123+andyziye@users.noreply.github.com>

Youngwb committed:
[BugFix] cache delta lake snapshot instead of table object (#54473)

banmoy committed:
[Enhancement] Return aborted reason when getting label/transaction state (#54472)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

sevev committed:
[BugFix] Fix primary key table add an visible rowset (#54491)

Signed-off-by: sevev <qiangzh95@gmail.com>

kobebryantlin0 committed:
[BugFix] fix `corrupted partition meta exception` when refreshing materialized views… (#54487)

Signed-off-by: AlgoLin <1354312240@qq.com>

decster committed:
[Feature] bulk load optimization(Part 3 Block merger) (#54285)

Signed-off-by: Binglin Chang <decstery@gmail.com>

LiShuMing committed:
[UT] Fix unstable mv test case testListPartitionTTLCondition3 (#54522)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

IrisesD committed:
[Doc] Fix #53995, #53996 and #53997 (#54046)

Signed-off-by: Cheng Ding <73172305+IrisesD@users.noreply.github.com>

dirtysalt committed:
[BugFix] remove unused code to avoid asan check failure (#54538)

edgarrmondragon committed:
[BugFix] Replace `sqlalchemy.util.raise` with Python 3 `raise` (#54518)

Signed-off-by: Edgar Ramírez-Mondragón <edgarrm358@gmail.com>

starrocks-xupeng committed:
[Enhancement] add timeout configs for star client (#54496)

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>

xiangguangyxg committed:
[Enhancement] Support fe starting with only image no bdb log (#54514)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

kevincai committed:
[Enhancement] allow multiple assignments to the same column in UPDATE (#54534)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

srlch committed:
[Feature] Support schema change for Generated Column in share data mode (#53526)

In pr #52407, Generated Column has been removed the restriction for shared data mode. In this pr, we support schema change operation for Generated Column in shared data mode.

Fixes #53527

Signed-off-by: srlch <linzichao@starrocks.com>

stdpain committed:
[Feature] support event scheduler for scan/exchange/agg (part 2) (#54338)

Signed-off-by: stdpain <drfeng08@gmail.com>

stdpain committed:
[BugFix] Fix wrong plan on limit with subquery TOP-N (#54507)

Signed-off-by: stdpain <drfeng08@gmail.com>

LiShuMing committed:
[BugFix] Add implicit cast rule for list partitino pruner with generated column (#54543)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

LiShuMing committed:
[BugFix] [UT] Fix test_transparent_mv_mysql test case (#54556)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

Youngwb committed:
[BugFix] Avoid reporting file not found errors during the BE upgrade process (#54560)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>

LiShuMing committed:
[BugFix] Disable stats collection for generated columns (#54537)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

kevincai committed:
[Enhancement] remove gcc14 detect in build-thirdparty script (#54576)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

wyb committed:
[Enhancement] Improve files() parquet orc column mismatch error message (#54571)

Signed-off-by: wyb <wybb86@gmail.com>

kevincai committed:
[Enhancement] patching flatbuffers CMakeLists.txt (#54579)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

LiShuMing committed:
[BugFix] Fix mv partition compensation bug for iceberg (#54581)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

Youngwb committed:
[UT] Fix Compilation error with clang (#54591)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>

murphyatwork committed:
[BugFix] fix resolve predicate columns of union operator (#54582)

Signed-off-by: Murphy <mofei@starrocks.com>

xiangguangyxg committed:
[Feature] Support restoring from a cluster snapshot for shared-data mode (part 4, use gtid to handle dirty tablet metadata) (#54484)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

ZiheLiu committed:
[Enhancement] add sv cbo_max_reorder_node_use_greedy and skip DP and greedy reorder without unknown stats (#54593)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

stdpain committed:
[BugFix] Fix UAF in shared UDF (#54592)

Signed-off-by: stdpain <drfeng08@gmail.com>

ZiheLiu committed:
[BugFix] Fix sequence of selection and input_chunk for streaming_agg_with_selection (#54595)

Signed-off-by: zihe.liu <ziheliu1024@gmail.com>

DanRoscigno committed:
[Doc] fix descriptions, add guide link (#54620)

wyb committed:
[BugFix] Fix decimal type merge in files() schema detection (#54583)

Signed-off-by: wyb <wybb86@gmail.com>

srlch committed:
[Feature] Support Cluster Snapshot Backup: support system table for cluster snapshot backup (part2) (#54508)

Signed-off-by: srlch <linzichao@starrocks.com>

stdpain committed:
[Doc] add doc for time_format (#54300)

Signed-off-by: stdpain <drfeng08@gmail.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

luohaha committed:
[Doc] update doc about shared-data mode ingestion rate limit (#54606)

Signed-off-by: luohaha <18810541851@163.com>
Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>
Co-authored-by: 絵空事スピリット <wanglichen@starrocks.com>

banmoy committed:
[Enhancement] Use bthread lock in merge commit rpc (#54575)

Signed-off-by: PengFei Li <lpengfei2016@gmail.com>

HangyuanLiu committed:
[BugFix] Fix warehouse exists check after modify system info meta bug (#54585)

tracymacding committed:
[Enhancement] Add skip_page_cache and skip_disk_cache options in session variables (#53824)

Signed-off-by: tracymacding <tracymacding@gmail.com>

stephen-shelby committed:
[Enhancement] adjust agg pushdown strategy for broadcast (#54572)

Signed-off-by: stephen <stephen5217@163.com>

sevev committed:
[Enhancement] Remove unnecessary frequent memory alloc and release requests (#54604)

Signed-off-by: sevev <qiangzh95@gmail.com>

stdpain committed:
[BugFix] Fix exception not clear in env (#54645)

Signed-off-by: stdpain <drfeng08@gmail.com>

wyb committed:
[Enhancement] Support read compressed csv in files() (#54626)

Signed-off-by: wyb <wybb86@gmail.com>

LiShuMing committed:
[BugFix] Only enable query context cache when there are more than one related mvs (#54627)

Signed-off-by: shuming.li <ming.moriarty@gmail.com>

kevincai committed:
[BugFix] fix HttpResultWriter ill-formatted json string (#54590)

Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

gengjun-git committed:
[Enhancement] Remove thrift_types_customize_impl.cpp from gitignore (#54564)

Signed-off-by: gengjun-git <gengjun@starrocks.com>

Commit:
[Doc] 3.1.17 release notes (#54684)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

gengjun-git committed:
[BugFix] Fixed compatibility bug when using jdk17 to compile into jdk8 bytecode (#54680)

## Why I'm doing:
Using the <maven.compiler.release>8</maven.compiler.release> configuration will use a too old version of the maven-compiler-plugin.

Use the latest version of maven-compiler-plugin to use the `javac -release` function

https://openjdk.org/jeps/247

## What I'm doing:

Fixes #https://github.com/StarRocks/StarRocksTest/issues/9011

Signed-off-by: gengjun-git <gengjun@starrocks.com>

zombee0 committed:
[Refactor]Abstract out the logictype-independent RawColumnReader for future optimization (#54675)

Signed-off-by: zombee0 <ewang2027@gmail.com>

trueeyu committed:
[Enhancement] Open or close interval need to be considered when using runtime filter to process zonemap. (#54687)

Signed-off-by: trueeyu <lxhhust350@qq.com>

Commit:
[Doc] 3.3.8 release notes (#54700)

Signed-off-by: 絵空事スピリット <wanglichen@starrocks.com>

sevev committed:
[BugFix] Reject table create request with time type in complex type (#54601)

Signed-off-by: sevev <qiangzh95@gmail.com>

dirtysalt committed:
[Enhancement] support `cache select` set var hint (#54697)

Signed-off-by: yanz <dirtysalt1987@gmail.com>

xiangguangyxg committed:
[BugFix] Remove sensitive messages in be log (#54677)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

dirtysalt committed:
[BugFix] avoid tte is null in `openInternal` in HiveMetaStoreClient (#54696)

Signed-off-by: yanz <dirtysalt1987@gmail.com>

stephen-shelby committed:
[Enhancement] statistics query error message retargeting from fe.out to fe.log (#54671)

Signed-off-by: stephen <stephen5217@163.com>

xiangguangyxg committed:
[Feature] Support restoring from a cluster snapshot for shared-data mode (part 5, support downloading image) (#54651)

Signed-off-by: xiangguangyxg <xiangguangyxg@gmail.com>

sevev committed:
[UT] Fix unstable ut (#54686)

Signed-off-by: sevev <qiangzh95@gmail.com>

HangyuanLiu committed:
[Feature] Support security view (#54458)

stephen-shelby committed:
[BugFix] don't remove partition predicate from predicate when bucket column includes partition column (#54679)

Signed-off-by: stephen <stephen5217@163.com>

maggie-zhu committed:
optimize the pinot sql transform in complex case

Signed-off-by: magzhu <magzhu@cisco.com>
